beacon_node/network/src/sync/network_context.rs

```diff
@@ -482,19 +575,20 @@ impl<T: BeaconChainTypes> SyncNetworkContext<T> {
                 request.remove();
                 Err(e.into())
             }
-        })
+        };
+        Some(((), resp))
```

Comment:
It feels a bit confusing that the tuple order (ID, Resp) is the opposite to the generic parameters (Resp, ID)